core/rawdb/accessors_chain.go

```diff
@@ -694,8 +694,21 @@ type receiptLogs struct {
 
 // DecodeRLP implements rlp.Decoder.
 func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
+	// Retrieve the entire receipt blob as we need to try multiple decoders
+	blob, err := s.Raw()
+	if err != nil {
+		return err
+	}
+	// Check to see if this is a celo dynamic fee receipt.
+	if types.IsCeloDynamicFeeReceipt(blob) {
+		var stored types.CeloDynamicFeeStoredReceiptRLP
+		err := rlp.DecodeBytes(blob, &stored)
```

Comment:
should you check the error here, instead of assigning maybe null to the r.logs?

Reply:
see - https://github.com/celo-org/op-geth/pull/242#discussion_r1777742847